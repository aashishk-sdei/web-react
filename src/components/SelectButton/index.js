import React from "react";
import PropTypes from "prop-types";
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
    selectButton: {
        background: "#FFFFFF",
        border: "1px solid #747578",
        boxSizing: 'border-box',
        borderRadius: "8px",
        width: 94,
        height: 40,
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        cursor: "pointer",
        marginRight: 8,
        color:"#747578",
        "&:hover": {
            background: "#DAE3EF",
            border:"1px solid #204A82",
            color:"#204A82"
        }
    },
    title: {
        fontStyle: "normal",
        fontWeight: "normal",
        fontSize: 14,
        whiteSpace: "nowrap",
        overflow: "hidden",
        textOverflow: "ellipsis",
        padding: 4
    },
    selectedButton: {
        background: "#DAE3EF",
        border:"1px solid #204A82",
        color:"#204A82"
    }
});

const SelectButton = ({ classes, title, select, handleSelect }) => {

    const handleClick = () => {
        handleSelect(title);
    }

    return (
        <div data-test="select-button" className={select ? `${classes.selectButton} ${classes.selectedButton}` : `${classes.selectButton}`} onClick={handleClick}>
            <div className={classes.title}>{title}</div>
        </div>
    )
}

SelectButton.propTypes = {
    title: PropTypes.string,
    select: PropTypes.bool,
    handleSelect: PropTypes.func
};

SelectButton.defaultProps = {
    title: "Title",
    select: false,
    handleSelect: undefined
}

export default withStyles(styles)(SelectButton);