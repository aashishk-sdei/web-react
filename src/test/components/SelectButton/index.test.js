import React from "react";
import SelectButton from "../../../components/SelectButton";
import { shallow } from "enzyme";

const title = "Title";
const handleSelect = undefined;
describe("SelectButton Component", () => {
  it("selectButton component renders properly", () => {
    const wrapper = shallow(<SelectButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it("accepts title props", () => {
    const wrapper = shallow(<SelectButton title={title} />);
    expect(wrapper.props().title).toEqual(title);
  });

  it("accepts handleSelect props", () => {
    const wrapper = shallow(<SelectButton handleSelect={handleSelect} />);
    expect(wrapper.props().handleSelect).toEqual(handleSelect);
  });
 
  it("select-button toBeTruthy", () => {
    const wrapper = shallow(<SelectButton/>);
    expect(wrapper.find("[data-test='select-button']")).toBeTruthy();
  });
});
