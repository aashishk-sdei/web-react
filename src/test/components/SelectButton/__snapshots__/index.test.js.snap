// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectButton Component selectButton component renders properly 1`] = `ShallowWrapper {}`;
